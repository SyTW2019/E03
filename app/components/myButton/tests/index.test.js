import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from 'react-testing-library';
import "jest-dom/extend-expect";
import renderer from "react-test-renderer";

import Button from '../index';

describe('<Button />', () => {

  afterEach(cleanup);

  it('render', () => {
    const div = document.createElement(div);
    ReactDOM.render(<Button/>,div);
  });

  it('should render a prop title', () => {
    const {getByTestId} = render(<Button title="Prueba"/>)
    expect(getByTestId('button')).toHaveTextContent("Prueba");
  });

  it('Matches snapshots', () => {
    const tree = renderer.create(<Button title="save"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
