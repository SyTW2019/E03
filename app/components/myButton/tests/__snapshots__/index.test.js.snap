// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Matches snapshots 1`] = `
<div>
  <button
    data-testid="button"
  >
    save
  </button>
</div>
`;
